### PR TITLE
[IMP] accounting/l10n_br: electronic invoicing & tax computation flow

### PR DESCRIPTION
**Pull request by vbe-odoo**
…n flow

With the addition of l10n_br_{avatax,edi,edi_sale}_services to the previous modules, now electronic invoices and tax computation for services can be created from Odoo through Avalara.

PR that includes these features: https://github.com/odoo/enterprise/pull/57868

This PR includes the necessary information to use these new modules - configuration and workflows.

G-Folder with new images: https://drive.google.com/drive/folders/1rCkyiYnWslzlQ7Oxb3IJC0-TeiWrQOwb?usp=sharing

### CODE REVIEW

**Review by samueljlieber — APPROVED** [review]
Hi @vbe-odoo, your improvements in this PR look good to me! I pushed up a small commit to add the images you used as well as to make a couple of small RST syntax corrections.

I believe this PR is in a good spot for the next reviewer :)

**Comment by vbe-odoo** [comment]
Hey @samueljlieber thanks for the changes and the review.
Just as a heads up the PRs that introduce this new feature were merged Today:
* https://github.com/odoo/odoo/pull/156017
* https://github.com/odoo/enterprise/pull/57868

And it will be available for customers starting next Monday, 22nd.
Thanks!
cc' @fvz-odoo @ren-odoo @masi-odoo

**Comment by vbe-odoo** [comment]
Hello @samueljlieber do you know if we can ping someone for this review?
Thanks!

**Comment by samueljlieber** [comment]
> Hello @samueljlieber do you know if we can ping someone for this review? Thanks!

Sorry about that @vbe-odoo, this PR got hung up and needed to be refreshed. The reviewers have been auto-tagged now 👍

File: content/applications/finance/fiscal_localizations/brazil.rst
Comment (dade-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:543]: Specify which document exactly (for example, "Correction letters are only available for electronic invoices for goods (NF-e)."

File: content/applications/finance/fiscal_localizations/brazil.rst
Comment (dade-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:562]: Specify which document exactly (for example, "Invalidate invoice number range documents are only available for electronic invoices for goods (NF-e)."

**Review by dade-odoo — CHANGES_REQUESTED** [review]
Hi @vbe-odoo and @samueljlieber - thanks for your work on this! I've made a few comments on improvements that I think could be made. I ask that you take a look at them and also at the screenshots. For any screenshot that doesn't add substantial value to the doc, it is better no to include it. Screenshots are burdensome to the code of the documentation and quickly become outdated as versions change. If a screenshot is absolutely necessary, that's fine, but otherwise we ask that you remove them. 

For the screenshots that are necessary, we then ask the same questions about any markups such as the red rectangles - are they absolutely necessary for the reader to understand the image? Stylistically, we would like to avoid these whenever possible :)

Thanks again for the contribution to the doc! I'm sure the Brazilian users will appreciate it!

File: content/applications/finance/fiscal_localizations/brazil.rst
Comment (samueljlieber) [reply content/applications/finance/fiscal_localizations/brazil.rst:51]: Thanks @dade-odoo, yes it appears Brazil Pix QR codes (l10n_br_pix) does not exist on saas-17.2. @vbe-odoo, can you confirm?

If so, I will plan to remove lines 54-56 on the saas-17.2 FWP.

**Comment by samueljlieber** [comment]
8ce66e6 Thank you for your thorough review @dade-odoo, I implemented each of your suggestions in 8ce66e6. As well as [commented on your point](https://github.com/odoo/documentation/pull/8784#discussion_r1583187616) about updated modules in saas-17.2.

File: content/applications/finance/fiscal_localizations/brazil.rst
Comment (vbe-odoo) [reply content/applications/finance/fiscal_localizations/brazil.rst]: Yup, you are right, I can see that l10n_br_pix was merged in 17.2 into l10n_br (https://github.com/odoo/odoo/pull/157324)

@samueljlieber can you please help me remove those lines (54-56 on the saas-17.2 FWP)?

Thanks!

**Comment by vbe-odoo** [comment]
Hello, @dade-odoo do you have more suggestions for this PR?
If not, do you think we can move forward with this? As we already have this feature merged and available for customers.

Thank you!
cc' @samueljlieber @ren-odoo @masi-odoo @fvz-odoo

**Comment by dade-odoo** [comment]
Hi @vbe-odoo @samueljlieber ! 
My apologies, I missed Sam's message on the 29th and then was out of office since the 1st. I'm still not convinced that the red boxes on the product configuration screenshot are necessary, but I'll let the final reviewer make that decision :)
Thanks again for adding this and I'll request the final review :)

**Review by dade-odoo — APPROVED** [review]

File: content/applications/finance/fiscal_localizations/brazil.rst
Comment (auva-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:107]:
```suggestion
   - :guilabel:`COFINS Details`: :guilabel:`Taxable, Not Taxable, Taxable with rate 0%, Exempt, Suspended`
   - :guilabel:`PIS Details`: :guilabel:`Taxable, Not Taxable, Taxable with rate 0%, Exempt, Suspended`
   - :guilabel:`CSLL Taxable`: If the company is subject to CSLL or not
```

File: content/applications/finance/fiscal_localizations/brazil.rst
Comment (auva-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:96]:
```suggestion
   - :guilabel:`Tax Regime`: Federal Tax Regime
   - :guilabel:`ICMS Taxpayer Type`: Indicates :guilabel:`ICMS regime`, :guilabel:`Exempt status`, or :guilabel:`Non-Taxpayer`
```

File: content/applications/finance/fiscal_localizations/brazil.rst
Comment (auva-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:81]: I would put the description or options after a semicolon instead of between brackets (it does't bother me that much here, but it does in other similar lists elsewhere in this doc so for the sake of consistency, I'd have them all the same)

File: content/applications/finance/fiscal_localizations/brazil.rst
Comment (auva-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:350]: Same as above regarding the brackets + add GUI labels where needed (see my other comment on the same topic above)

File: content/applications/finance/fiscal_localizations/brazil.rst
Comment (auva-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:435]:
```suggestion
:guilabel:`Send & Print` button in the upper left corner. In the pop-up that appears, click on
```

File: content/applications/finance/fiscal_localizations/brazil.rst
Comment (auva-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:443]:
```suggestion
- :guilabel:`Customer`, with all the customer information
```

File: content/applications/finance/fiscal_localizations/brazil.rst
Comment (auva-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:444]:
```suggestion
- :guilabel:`Payment Method: Brazil`: how the invoice is planned to be paid
```

File: content/applications/finance/fiscal_localizations/brazil.rst
Comment (auva-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:506]:
```suggestion
E-invoices for goods (NF-e)
```

File: content/applications/finance/fiscal_localizations/brazil.rst
Comment (auva-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:519]:
```suggestion
E-invoices for services (NFS-e)
```

**Review by auva-odoo — CHANGES_REQUESTED** [review]
Hello @vbe-odoo ! Thank you for your work :slightly_smiling_face: . In addition to the comments and suggestions I added, here are some general remarks:
- Could you please remove the last sentence in your commit message (the one referring to the Google Drive)? This is more of an internal note and doesn't need to be in there.
- I agree with Dallas that the red boxes are unnecessary in this case, especially since they are well cropped and don't show much outside of what you're referring to. I would therefore remove the markup from them.
- Could we please use AvaTax everywhere (I found a mix of AvaTax and Avatax)
- I would put field descriptions or options in bullet list items after a semicolon instead of 
between brackets, and use guilabel formatting for options. I added a comment where this happens, but just in case, here’s a general comment 🙂
Let me know if you have any questions!

**Review by auva-odoo — APPROVED** [review]
Hi @samueljlieber thank you! Since I'll be off next week, and this PR is nearly ready, I don't want to keep you waiting until I return to merge it. I'll go ahead and approve it now, and you can r+ it once the remaining changes are addressed. Let me know if this works for you!

**Comment by ren-odoo** [comment]
Hi @samueljlieber indeed VBE is OOO so let's move forward with this PR then. 

Thank you.

**Comment by samueljlieber** [comment]
Thank you all! 🚀

@robodoo r+

**Pull request closed without merging**
